### PR TITLE
Update DoctrineOrmMapper.php

### PR DESCRIPTION
**Pull request by n0-m4d**
while implementing collections using ORM i came accross the missing implementation for `DoctrineOrmMapper:prepareObject()`... well, here is mine.

### CODE REVIEW

File: src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php [hunk 1]
```diff
@@ -37,6 +37,36 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
...
+    {
+        $object = parent::prepareObject($type);
+
+        if (null == $parent){
+            throw new RuntimeException('You need a parent to create new objects');
+        }
+
+        /** @var \Doctrine\ORM\Mapping\ClassMetadata $metaData */
+        $metaData = $this->om->getClassMetaData(get_class($object));
+        $metaDataParent = $this->om->getClassMetaData(get_class($parent));
```
Comment (dbu) [inline src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php]: i would use Doctrine\Common\Util\ClassUtils::getRealClass for this, as you do not really need the metadata.

File: src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php [hunk 2]
```diff
@@ -37,6 +37,36 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
...
+
+        foreach ($metaData->associationMappings as $mapping)
```
Comment (dbu) [inline src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php]: i would move the code where you search for the parent field into a method `findParentMapping`. that way its easier to customize it when needed.

Comment (n0-m4d) [inline src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php]: how would you parametrize `findParentMapping`? i was thinking about to use `$metaData` as a parameter but that might be a bad idea if someone wants to override this method although i would have to retrieve `$metaData`twice. Should I use `$object` or just the string from `ClassUtils:getRealClass`? hard for me to imagine which cases might occur that require a customized version of `findParentMapping`...

Comment (dbu) [inline src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php]: the use case i think of is that e.g. my class has references to objects of the same class once as parent and once as children. then i would need to have custom logic as the generic logic fails. 

i would pass $object and $metadata. we are in an orm context, so it does not hurt to have it.

**Pull request merged**